test/extensions/structural_identifiability.jl

```diff
@@ -0,0 +1,293 @@
+### Fetch Packages ###
+
+using Catalyst, Test
+using StructuralIdentifiability
+
+
+### Helper Function ###
+
+# Converts the output dicts from StructuralIdentifiability functions from "weird symbol => stuff" to "symbol => stuff" (the output have some strange meta data which prevents equality checks, this enables this).
+# Structural identifiability also provides variables like x (rather than x(t)). This is a bug, but we have to convert to make it work (now just remove any (t) to make them all equal).
+function sym_dict(dict_in)
+    dict_out = Dict{Symbol,Any}()
+    for key in keys(dict_in)
+        sym_key = Symbol(key)
+        sym_key = Symbol(replace(String(sym_key), "(t)" => ""))
+        dict_out[sym_key] = dict_in[key]
+    end    
+    return dict_out
+end
+
+
+### Run Tests ###
+
+# Tests for Goodwin model (model with both global, local, and non identifiable components).
+# Tests for system using Catalyst function (in this case, Michaelis-Menten function)
+let 
+    # Identifiability analysis for Catalyst model.
+    goodwind_oscillator_catalyst = @reaction_network begin
+        (mmr(P,pₘ,1), dₘ), 0 <--> M
+        (pₑ*M,dₑ), 0 <--> E
+        (pₚ*E,dₚ), 0 <--> P
+    end
+    gi_1 = assess_identifiability(goodwind_oscillator_catalyst; measured_quantities=[:M])
+    li_1 = assess_local_identifiability(goodwind_oscillator_catalyst; measured_quantities=[:M])
+    ifs_1 = find_identifiable_functions(goodwind_oscillator_catalyst; measured_quantities=[:M])
+
+    # Identifiability analysis for Catalyst converted to StructuralIdentifiability.jl model.
+    si_catalyst_ode = make_si_ode(goodwind_oscillator_catalyst; measured_quantities=[:M])
+    gi_2 = assess_identifiability(si_catalyst_ode)
+    li_2 = assess_local_identifiability(si_catalyst_ode)
+    ifs_2 = find_identifiable_functions(si_catalyst_ode)
+
+    # Identifiability analysis for StructuralIdentifiability.jl model (declare this overwrites e.g. X2 variable etc.).
+    goodwind_oscillator_si = @ODEmodel(
+        M'(t) = pₘ / (1 + P(t)) - dₘ*M(t), 
+        E'(t) = -dₑ*E(t) + pₑ*M(t),
+        P'(t) = -dₚ*P(t) + pₚ*E(t),
+        y1(t) = M(t)
+    )
+    gi_3 = assess_identifiability(goodwind_oscillator_si)
+    li_3 = assess_local_identifiability(goodwind_oscillator_si)
+    ifs_3 = find_identifiable_functions(goodwind_oscillator_si)
+
+    # Check outputs.
+    @test sym_dict(gi_1) == sym_dict(gi_2) == sym_dict(gi_3)
+    @test sym_dict(li_1) == sym_dict(li_2) == sym_dict(li_3)
+    @test length(ifs_1) == length(ifs_2) == length(ifs_3)     
```

Comment:
Can we hardcode what the answers should be and test against that too?

Reply:
For all 3, or just the last one? I think the last one could technically be a bit stochastic what expression it produces.

For the others we could.

Reply:
added